implementation/common/src/test/java/io/smallrye/jwt/util/GenerateSecretFromKeyEncryptionAlgoTest.java

```diff
@@ -0,0 +1,40 @@
+package io.smallrye.jwt.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import javax.crypto.SecretKey;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import io.smallrye.jwt.algorithm.KeyEncryptionAlgorithm;
+
+@RunWith(Parameterized.class)
+public class GenerateSecretFromKeyEncryptionAlgoTest {
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { "PBES2-HS256+A128KW", 128 }, { "PBES2-HS384+A192KW", 192 }, { "PBES2-HS512+A256KW", 256 }
```

Comment:
These should be `A128KW` etc. `HS` in these PBES algos does not reflect the fact the password entered by the user may be different size... The scheme should be `AES`, please also see below for the proposal on having a map inside `KeyUtils` to avoid us having to parse the enum names, etc